src/ducks/settings/CategoryAlerts/CategoryAlertEditModal.jsx

```diff
@@ -137,6 +137,82 @@ const fieldSpecs = {
   }
 }
 
+const EditionModal = props => {
+  const {
+    fieldSpecs,
+    fieldLabels,
+    fieldOrder,
+    initialDoc,
+    modalTitle,
+    okButtonLabel,
+    cancelButtonLabel,
+    onEdit,
+    onDismiss
+  } = props
+  const [doc, setDoc] = useState(initialDoc)
+  const [choosing, setChoosing] = useState(null)
+
+  const handleChoosingCancel = () => {
+    setChoosing(null)
+  }
+
+  const handleChangeField = (name, val) => {
+    const updater = fieldSpecs[name].updater
+    const updatedDoc = updater(doc, val)
+    setDoc(updatedDoc)
+  }
+
+  const handleRequestChooseField = name => {
+    const fieldSpec = fieldSpecs[name]
+    setChoosing({
+      type: fieldSpec.type,
+      value: fieldSpec.getValue(doc),
+      onSelect: val => {
+        setChoosing(null)
+        handleChangeField(name, val)
+      },
+      onCancel: handleChoosingCancel
+    })
+  }
+
+  const handleConfirmEdit = () => {
+    onEdit(doc)
+  }
+
+  return (
+    <Modal title={modalTitle} mobileFullscreen={true} dismissAction={onDismiss}>
+      <Stepper
+        showPercentage={false}
+        currentIndex={choosing ? 1 : 0}
```

Comment:
Nit: Even if they look self explanatory at first sight, these 1 and 0 look a lot like magic values. I'm afraid we will not be comfortable with it in the future. Maybe we should put it in variables with clear names or add a comment?